src/OOPMacroImpl.jl

```diff
@@ -95,15 +95,16 @@ macro class(args...)
           end"""
 
     # this allows calling functions on the class..
+    clsFnNameList = join(map(fn->":$(getFnName(fn, withoutGeneric=true)),", collect(values(methods))),"")
     dotAccStr = """
-              function Base.getproperty(self::$clsName, name::Symbol)
-                  try
-                    getfield(self, name)
-                  catch
-                    (args...; kwargs...)->eval(:(\$name(\$self, \$args...; \$kwargs...)))
-                  end
-              end
-              """
+        function Base.getproperty(self::$clsName, name::Symbol)
+            if isdefined(self, name) || name ∉ ($clsFnNameList)
```

Comment:
Why do we nee clsFnNameList here? I think when isdefined(self, name), the best we could do is to try to call the function, so why do we still need to check if the name in the function list?